common4j/src/main/com/microsoft/identity/common/java/opentelemetry/OTelUtility.java

```diff
@@ -51,6 +55,36 @@ public static Span createSpan(@NonNull final String name) {
         return span;
     }
 
+    /**
+     * Creates a span from a parent Span Context (with shared basic attributes).
+     **/
+    @NonNull
+    public static Span createSpanFromParent(@NonNull final String name,
+                                            @Nullable final SpanContext parentSpanContext) {
+        final String methodTag = TAG + ":createSpanFromParent";
+
+        if (parentSpanContext == null) {
+            Logger.verbose(methodTag, "parentSpanContext is NULL. Creating span without parent.");
+            return createSpan(name);
+        }
+
+        if (!parentSpanContext.isValid()) {
+            Logger.warn(methodTag, "parentSpanContext is INVALID. Creating span without parent.");
+            return createSpan(name);
+        }
+
+        final Tracer tracer = GlobalOpenTelemetry.getTracer(TAG);
+        final Span span = tracer.spanBuilder(name)
+                .setParent(Context.current().with(Span.wrap(parentSpanContext)))
+                .startSpan();
+
+        if (parentSpanContext instanceof SerializableSpanContext) {
```

Comment:
else, is it an unexpected scenario? Do we want to add a log for such a scenario?

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1926/commits/b2095cb6271fdac7c4a150253e2ea36bdc661390